karapace/schema_reader.py

```diff
@@ -254,105 +266,122 @@ def handle_messages(self):
                 if self.ready and add_offsets:
                     self.offset_watcher.offset_seen(self.offset)
 
-    def handle_msg(self, key: dict, value: dict):
```

Comment:
`value: dict` was not correct. For instance, a hard deleted schema has an empty value

- [Here it is set to None](https://github.com/aiven/karapace/blob/master/karapace/schema_registry_apis.py#L603-L605)
- [Here it translated to ""](https://github.com/aiven/karapace/blob/4959ad42d281859ac0efe03566141595c6fb28a2/karapace/schema_registry_apis.py#L272)

Then while [decoding the empty string](https://github.com/aiven/karapace/blob/4959ad42d281859ac0efe03566141595c6fb28a2/karapace/schema_reader.py#L242-L248), it should fail, because an empty string is not valid json:

```python
>>> import json
>>> json.loads('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

So the `value` can be `None`.